libraries/AP_DDS/README.md

```diff
@@ -164,135 +164,196 @@ Next, follow the associated section for your chosen transport, and finally you c
+response:
+ardupilot_msgs.srv.ArmMotors_Response(result=True)
+```
  
-## Contributing to AP_DDS library
+## Contributing to `AP_DDS` library
+
 ### Adding DDS messages to Ardupilot
 
 Unlike the use of ROS 2 `.msg` files, since Ardupilot supports native DDS, the message files follow [OMG IDL DDS v4.2](https://www.omg.org/spec/IDL/4.2/PDF).
 This package is intended to work with any `.idl` file complying with those extensions.
 
 Over time, these restrictions will ideally go away.
 
-To get a new IDL file from ROS2, follow this process:
-```console
+To get a new IDL file from ROS 2, follow this process:
+
+```bash
 cd ardupilot
 source /opt/ros/humble/setup.bash
+
 # Find the IDL file
 find /opt/ros/$ROS_DISTRO -type f -wholename \*builtin_interfaces/msg/Time.idl
+
 # Create the directory in the source tree if it doesn't exist similar to the one found in the ros directory
 mkdir -p libraries/AP_DDS/Idl/builtin_interfaces/msg/
+
 # Copy the IDL
 cp /opt/ros/humble/share/builtin_interfaces/msg/Time.idl libraries/AP_DDS/Idl/builtin_interfaces/msg/
+
 # Build the code again with the `--enable-dds` flag as described above
 ```
 
+### Rules for adding topics and services to `dds_xrce_profile.xml`
+
+Topics and services available from `AP_DDS` are automatically mapped into ROS 2
+provided a few rules are followed when defining the entries in
+`dds_xrce_profile.xml`.
+
+#### ROS 2 message and service interface types
+
+ROS 2 message and interface defintions are mangled by the `rosidl_adapter` when
+mapping from ROS 2 to DDS to avoid naming conflicts in the C/C++ libraries.
+The ROS 2 object `namespace::Struct` is mangled to `namespace::dds_::Struct_`
+for DDS. The table below provides some example mappings: 
+
+| ROS 2 | DDS |
+| --- | --- |
+| `rosgraph_msgs::msg::Clock` | `rosgraph_msgs::msg::dds_::Clock_` |
+| `sensor_msgs::msg::NavSatFix` | `sensor_msgs::msg::dds_::NavSatFix_` |
+| `ardupilot_msgs::srv::ArmMotors_Request` | `ardupilot_msgs::srv::dds_::ArmMotors_Request_` |
+| `ardupilot_msgs::srv::ArmMotors_Response` | `ardupilot_msgs::srv::dds_::ArmMotors_Response_` |
+
+Note that a service interface always requires a Request / Response pair.
+
+#### ROS 2 topic and service names
+
+The ROS 2 design article: [Topic and Service name mapping to DDS](https://design.ros2.org/articles/topic_and_service_names.html) describes the mapping of ROS 2 topic and service
+names to DDS. Each ROS 2 subsytem is provided a prefix when mapped to DDS.
+The request / response pair for services require an additional suffix.
+
+| ROS 2 subsystem | DDS Prefix | DDS Suffix |
+| --- | --- | --- |
+| topics | rt/ | |
+| service request | rq/ | Request |
+| service response | rr/ | Reply |
+| service | rs/ | |
+| parameter | rp/ | |
+| action | ra/ | |
+
+The table below provides example mappings for topics and services 
+
+| ROS 2 | DDS |
+| --- | --- |
+| ap/clock | rt/ap/clock |
```

Comment:
Thanks for adding these examples. Although I was familiar with the `rt` prefix, the others are new to me. Good job chasing these down.

Reply:
Indeed..really helpful examples here @srmainwaring